flang/include/flang/ISO_Fortran_binding.h

```diff
@@ -155,6 +162,19 @@ typedef struct CFI_cdesc_t {
 #else
   CFI_dim_t dim[]; /* must appear last */
 #endif
+
+#ifdef __cplusplus
+  RT_API_ATTRS inline bool HasAddendum() const {
+    return extra & _CFI_ADDENDUM_FLAG;
+  }
+  RT_API_ATTRS inline void SetHasAddendum() { extra |= _CFI_ADDENDUM_FLAG; }
+  RT_API_ATTRS inline int GetAllocIdx() const {
+    return ((int)extra >> _CFI_ADDENDUM_BITS);
```

Comment:
You don't need the old-style cast to `(int)` and you should also mask the field after the shift to be ready for additional fields to be added.

Reply:
So how many bits are we keeping for allocator index?

Reply:
How about 3 bits?

Reply:
I think this is reasonable. I'll make the update.